odl/solvers/iterative/statistical.py

```diff
@@ -121,6 +138,131 @@ def mlem(op, x, data, niter=1, noise='poisson', callback=None):
         raise RuntimeError('unknown noise model')
 
 
+def osmlem(op, x, data, niter=1, noise='poisson', callback=None, **kwargs):
+    """Ordered Subsets Maximum Likelihood Expectation Maximation algorithm.
+
+    Attempts to solve::
+
+        max_x L(data | x)
+
+    where ``L(data, | x)`` is the likelihood of ``data`` given ``x``. The
+    likelihood depends on the forward operators ``op[0], ..., op[n-1]`` such
+    that (approximately)::
+
+        op[i](x) = data[i]
+
+    With the precise form of *approximately* is determined by ``noise``.
+
+    Parameters
+    ----------
+    op : sequence of `Operator`
+        Forward operators in the inverse problem.
+    x : ``op.domain`` element
+        Vector to which the result is written. Its initial value is
+        used as starting point of the iteration, and its values are
+        updated in each iteration step.
+    data : sequence of ``op.range`` `element-like`
+        Right-hand sides of the equation defining the inverse problem.
+    niter : int, optional
+        Number of iterations.
+    noise : {'poisson'}, optional
+        Noise model determining the variant of MLEM.
+        For ``'poisson'``, the initial value of ``x`` should be
+        non-negative.
+    callback : callable, optional
+        Function called with the current iterate after each iteration.
+
+    Other Parameters
+    ----------------
+    sensitivities : float or ``op.domain`` `element-like`, optional
+        Usable with ``noise='poisson'``. The algorithm contains a ``A^T 1``
+        term, if this parameter is given, it is replaced by it.
+        Default: ``op[i].adjoint(op[i].range.one())``
+
+    Notes
+    -----
+    Given a forward models :math:`A_i`, data :math:`g_i`, :math:`i = 1, ..., M`
+    and noise model :math:`X`, the algorithm attempts find an :math:`x` that
+    maximizes:
+
+    .. math::
+
+        \prod_{i=1}^M P(g_i | g_i \\text{ is } X(A_i(x)) \\text{ distributed})
+
+    where the expectation of :math:`X(A_i(x))` satisfies
+
+    .. math::
+
+        \\mathbb{E}(X(A_i(x))) = A_i(x)
```

Comment:
Could you be a bit more precise here? I don't see where an expectation is maximized here.